src/main/java/org/gridsuite/gateway/GatewayConfig.java

```diff
@@ -0,0 +1,147 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.gateway;
+
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.nimbusds.openid.connect.sdk.claims.IDTokenClaimsSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gateway.filter.GatewayFilterChain;
+import org.springframework.cloud.gateway.filter.GlobalFilter;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import java.net.*;
+import com.nimbusds.jose.*;
+import com.nimbusds.jwt.*;
+import com.nimbusds.oauth2.sdk.id.*;
+import com.nimbusds.openid.connect.sdk.validators.*;
+
+/**
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+@Configuration
+@EnableConfigurationProperties(UriConfiguration.class)
+public class GatewayConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GatewayConfig.class);
+
+    private final GatewayService gatewayService;
+    private boolean ignoreTokenValidation;
+
+    @Value("${allowed-issuers}")
+    private List<String> allowedIssuers;
+
+    public GatewayConfig(GatewayService gatewayService, UriConfiguration uriConfiguration) {
+        this.gatewayService = gatewayService;
+        this.ignoreTokenValidation = uriConfiguration.getIgnoreTokenValidation();
+    }
+
+    @Component
+    public class TokenValidatorGlobalPreFilter implements GlobalFilter {
+        @Override
+        public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+            if (ignoreTokenValidation) {
+                return chain.filter(exchange);
+            }
+            LOGGER.debug("checking authorization");
+            List<String> ls = exchange.getRequest().getHeaders().get("Authorization");
+            if (ls == null) {
+                throw new GatewayException("Authorization header is required");
+            }
+
+            LOGGER.debug("checking issuer");
+            String authorization = ls.get(0);
+            String token = authorization.split(" ")[1];
+            DecodedJWT jwt = com.auth0.jwt.JWT.decode(token);
+            if (!allowedIssuers.contains(jwt.getIssuer())) {
+                throw new GatewayException(jwt.getIssuer() + " Issuer is not in the issuers white list");
```

Comment:
We need more advanced checks here because for azure the issuer is not fixed. Maybe startsWith is enough ?

Reply:
the whitelist will be configured in a config file so i think there's no problem here

Reply:
No because with azure you don't know the issuer. Depending on the account it can be anything, but it will always be something like "https://login.microsoftonline.com/XXX/v2.0" so always starting with https://login.microsoftonline.com

Reply:
okay